src/basis.jl

```diff
@@ -369,23 +388,24 @@ must be manually instantiated if more fine-grained control is desired.
    is used to determine which atoms impact to the bond's environment.
 - `λₙ::AbstractFloat`: 
 - `λₗ::AbstractFloat`:
+- `species::Union{nothing, Vector{AtomicNumber}}`: A set of species of a system
 
 # Returns
 - `basis::SymmetricBasis`: ACE basis entity for modelling the specified interaction. 
 
 """
 function off_site_ace_basis(ℓ₁::I, ℓ₂::I, ν::I, deg::I, b_cut::F, e_cutₒᵤₜ::F=5.;
-    λₙ::F=.5, λₗ::F=.5, symfix=true) where {I<:Integer, F<:AbstractFloat}
+    λₙ::F=.5, λₗ::F=.5, symfix=true, species = nothing) where {I<:Integer, F<:AbstractFloat}
     # WARNING symfix might cause issues when applied to interactions between different species.
     # It is still not clear how appropriate this non homo-shell interactions.
     @static if SYMMETRY_FIX_ENABLED
-        if symfix
-            basis = _off_site_ace_basis_sym(ℓ₁, ℓ₂, ν, deg, b_cut, e_cutₒᵤₜ; λₙ=λₙ, λₗ=λₗ)
+        if symfix && length(species) <= 1
```

Comment:
There is an bug here. By default `species` will be set to `nothing`. If you try to evaluate `length(nothing)` you get an error which is as expected behavior. I notice that you have defined `Base.length(a::Nothing) = 0` elsewhere in the code but this can cause unexpected and very hard to diagnose issues. A better solution would be as follows:
```suggestion
        if symfix && (isnothing(species) && length(species) ≠ 0)
```
As `&&`  is a short-circuiting operator the length check will only be performed it is not nothing. Also it might be a good idea to type guard the species input.

Reply:
I change it a bit (`&&` \> `||`). If species = nothing, then the `&&` check will go into the length check and give an error. I do think this is just a typo.

Reply:
Right I think I got the expected behavior swapped around. I misunderstood this to be `true` if `symfix` was enabled `and` there was a species list provided. My mistake. In that case use your suggested correction.

Reply:
When we turn `symfix = true`, it means that we want to use a bond-refl-sym basis. But up to now I cannot say 100% that our codes work for multi-species basis so we should only use `_off_site_ace_basis_sym(...)` when `species = nothing` or only has only one element.

Hence we will need a `(isnothing(species) || length(species) <= 1)` rather than `(isnothing(species) && length(species) ≠ 0)`